### PR TITLE
Newest update from video_streaming group

### PR DESCRIPTION
**Pull request by Mikkelmmc**
## Description

This is a pull request from the video streaming group, where we worked on the explore screen and the navigation bar in the bottom

## Changes

Created:
- Navigation bar (bottom)
- Explore screen
- Explore course card

## Related Issues

No. 

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

## Screenshots (if applicable)

Skærmbillede 2023-10-06 kl  11 52 04

Skærmbillede 2023-10-06 kl  11 52 23

Skærmbillede 2023-10-06 kl  11 51 31

## Notes for Reviewers

It's mostly changes in the design, it is working on both IOS and Android

### CODE REVIEW

**Comment by SlovenlyTF** [comment]
Husk at bruge den pull request template for projectet :D
https://github.com/Educado-App/resources/blob/main/PULL_REQUEST_TEMPLATE.md

**Comment by SlovenlyTF** [comment]
Er den her pull request en færdig PBI, eller bare kode der virker til showcase mandag?

**Comment by Jatewo** [comment]
Please update the pull request to use the pull request template from the resource repo.

**Comment by Jatewo** [comment]
Also remember that you can request reviews from others 👍 

In case you are not ready for reviewing, please convert the PR to a draft 😄

**Comment by Mikkelmmc** [comment]
> * Navbar
> * Explorar screen

**Comment by Mikkelmmc** [comment]
* [Explore Page + Navbar]
    * We have improved the Navbar as well as the explore page in the mobile app to look like their Figma prototypes. 
🔧 Improvements
* [Improvement Navbar]
    * We fixed the Navbar to look as described in its Figma prototype, and have navigation work properly for existing modules.
* [Improvement Explore Page]
    * We fixed the explore page to look as described in its Figma prototype, and it now has courses to explore within it, that can be folded out to explore further details. 
🐛 Bug Fixes
* [Issue or Bug Name]
    * No particular bugs, but rather just an improvement to what was before. 
📖 Documentation
* “”

**Comment by JackSMunn** [comment]
Not using template for PR's. Please do so. Converting to draft to not clutter PRs that need immediate attention.  Reopen when ready.

File: eml/App.js [hunk 1]
```diff
@@ -98,116 +99,31 @@ function LoginStack() {
...
-                type="material-community"
-                color="black"
-              />
-            );
-          },
-        }}
-      /> */}
-    </Tab.Navigator>
-  );
-}
```
Comment (JackSMunn) [inline eml/App.js:186]: What is the reason for deleting the whole stack?

File: eml/api/api.js [hunk 2]
```diff
@@ -42,15 +40,21 @@ export const getCourseWithAuth = async (courseId) => {
 
 export const getCourses = async () => {
   // TODO: add bearer token to request header and omit /public
-  const res = await axios.get(url + '/api/public/courses');
+  const res = await axios.get(url + '/api/courses');
+  return res.data;
```
Comment (JackSMunn) [inline eml/api/api.js:44]: Microsoft convention requires endpoint to be singular. (not plural) 
So /api/course

File: eml/api/api.js [hunk 3]
```diff
@@ -42,15 +40,21 @@ export const getCourseWithAuth = async (courseId) => {
...
+};
+
+export const getHome = async () => {
+  const res = await axios.get(url + '/api/course/getHome');
```
Comment (JackSMunn) [inline eml/api/api.js:48]: Microsoft convention for endpoint says that http request types are not used in endpoints. 
So /api/course/home

File: eml/components/courses/courseCard/CourseCard.jsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+export default function CourseCard({ course, downloadState }) {
+    const navigation = useNavigation();
+    
+    let [fontsLoaded] = useFonts({
+        VarelaRound_400Regular,
+      });
+    
+      if (!fontsLoaded) {
+        return AppLoading;
```
Comment (JackSMunn) [inline eml/components/courses/courseCard/CourseCard.jsx:15]: Do i understand it correctly, that you just return program into an idling animation if the font is not loading?

File: eml/components/courses/courseHeader/CourseProgress.jsx [hunk 5]
```diff
@@ -9,26 +9,42 @@ export default function CourseProgress({ fracTop, fracBot }) {
...
+    position: 'absolute'
   },
   fracStyle: {
     fontSize: 20,
-    fontWeight: 'bold'
+    fontWeight: 'bold',
+    color: 'black',
+    position: 'absolute',
+    alignSelf: 'center',
+    bottom: -2,
```
Comment (JackSMunn) [inline eml/components/courses/courseHeader/CourseProgress.jsx:48]: This whole stylesheet block needs to be reworked to use NativeWind and not stylesheets, as agreed upon by integration team.

File: eml/components/courses/courseHeader/CourseTitleIcon.jsx [hunk 6]
```diff
@@ -3,49 +3,62 @@ import { StyleSheet, View, Image } from 'react-native'
...
   CourseTitleIcon.propTypes = {
     title: PropTypes.string.isRequired,
-    courseIcon: PropTypes.string
+    courseIcon: PropTypes.string,
+    category: PropTypes.string.isRequired,
+    progress: PropTypes.string.isRequired
   }
 
   let [fontsLoaded] = useFonts({
     VarelaRound_400Regular
```
Comment (JackSMunn) [inline eml/components/courses/courseHeader/CourseTitleIcon.jsx:17]: Correct font? We're using Montserrat

File: eml/components/courses/courseHeader/CourseTitleIcon.jsx [hunk 7]
```diff
@@ -3,49 +3,62 @@ import { StyleSheet, View, Image } from 'react-native'
...
+          style={{width: Dimensions.get('window').width*0.25, height: Dimensions.get('window').width*0.25 }}
+          /> 
+      </View>
+      <View className="pt-1 pl-3 pr-3" style={{width: Dimensions.get('window').width*0.6}}>
         <Text
           numberOfLines={1}
           ellipsizeMode={'tail'}
-          style={{ fontSize: 30, fontFamily: 'VarelaRound_400Regular', color: 'rgb(75,85,99)' }}
+          style={{fontSize: 23, fontFamily: 'VarelaRound_400Regular', color: 'rgb(255,0,0)'}}
         >
```
Comment (JackSMunn) [inline eml/components/courses/courseHeader/CourseTitleIcon.jsx:35]: Wrong font again

File: eml/components/courses/courseHeader/CourseTitleIcon.jsx [hunk 8]
```diff
@@ -3,49 +3,62 @@ import { StyleSheet, View, Image } from 'react-native'
...
-          /> :
-          <Image source={require('../../../assets/loadingImage.png')}
-            style={{ width: 50, height: 50 }}
-            className="rounded-xl"
-          />
-        }
+        <Text
+          numberOfLines={1}
+          ellipsizeMode={'tail'}
+          style={{fontSize: 16, fontFamily: 'VarelaRound_400Regular', color: 'rgb(0,0,0)'}}
```
Comment (JackSMunn) [inline eml/components/courses/courseHeader/CourseTitleIcon.jsx:41]: Wrong font

File: eml/components/courses/courseHeader/CourseTitleIcon.jsx [hunk 9]
```diff
@@ -3,49 +3,62 @@ import { StyleSheet, View, Image } from 'react-native'
...
+    height: Dimensions.get('window').width*0.32,
+    flexDirection: 'row',
     justifyContent: 'center',
-    alignItems: 'center'
+    alignItems: 'center',
+    backgroundColor: 'rgb(255,255,255)',
+    borderBottomLeftRadius: 20,
+    borderTopLeftRadius: 20,
+    overflow: 'hidden'
   },
```
Comment (JackSMunn) [inline eml/components/courses/courseHeader/CourseTitleIcon.jsx:63]: Again, stylesheets, integration team agreed to use NativeWind.

File: eml/App.js (hunk 1 above)
Comment (Mikkelmmc) [reply eml/App.js:186]: Because we made a 'NavBar' component instead.

File: eml/components/courses/section/SectionCard.jsx [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+                                right: 10,
+                            }} />
+                        </TouchableOpacity>
+                    </View>
+                </Collapsible>
+
+            </Pressable>
+        </View>
+    );
+}
```
Comment (JackSMunn) [inline eml/components/courses/section/SectionCard.jsx:108]: Whole file uses inline-styling. Please use NativeWind as the integration team agreed upon.

**Review by JackSMunn — CHANGES_REQUESTED** [review]
Looks cool, I love when you can see that something is happening.

Please be descriptive in your changes. E.g. "changes to model" Could be "changes made to x,y,z in user model in file: user.js

Please refactor to NativeWind

File: eml/components/easyDynComponents/EasyDynamicList.jsx [hunk 11]
```diff
@@ -5,23 +5,23 @@ import SectionItem from '../courses/courseBody/SectionItem'
...
+                            //key={index}
                             courseId={course.id}
-                            title={item.title}
-                            index={index + 1}
+                            title={item}
+                            //index={index + 1}
                         />
                     )
-                })}
+                })*/}
```
Comment (JackSMunn) [inline eml/components/easyDynComponents/EasyDynamicList.jsx:24]: Expect to use this code? If answer is no, delete dead code

File: eml/components/explore/AccesCourseButton.jsx [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+          fontWeight: 'bold',
+        }}
+      >
+        
+        Acessar curso
+      </Text>
+    </Pressable>
+  );
+};
+
```
Comment (JackSMunn) [inline eml/components/explore/AccesCourseButton.jsx:32]: Refactor to use NativeWInd

File: eml/components/explore/CardLabel.jsx [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+        <View style={{
+            flexDirection: 'row',
+            alignItems: 'center',
+            justifyContent: 'flex-start',
+        }}>
+            <MaterialCommunityIcons name={icon} size={13} color={color} />
+            <Text style={{ paddingLeft: 3, fontSize: 13, color: color }} >{displayTitle}</Text>
+        </View>
+    );
+}
```
Comment (JackSMunn) [inline eml/components/explore/CardLabel.jsx:18]: Refactor to use NativeWind

**Review by JackSMunn — CHANGES_REQUESTED** [review]
I love when something tangible gets produced and it's not just pure backend stuff.
I have some comments regarding documentation, styles and font.

Did you document enough? I understand the code, but I'm not sure other people will in the future.

Refactor all styles to use NativeWind
Change font, i know Jacob & August have created an easy way to do it. 
Be more specific in changes section of pull request template. "changes to what? where, why"

File: eml/api/api.js (hunk 3 above)
Comment (Jatewo) [reply eml/api/api.js:48]: +1

File: eml/api/api.js (hunk 2 above)
Comment (Jatewo) [reply eml/api/api.js:44]: +1

File: eml/api/api.js [hunk 14]
```diff
@@ -61,6 +65,24 @@ export const getPresignedUrl = async (component_id) => {
   return res.data;
 };
 
+
+//CREATE FOR TESTING: BUT IT WORKS
```
Comment (Jatewo) [inline eml/api/api.js:69]: Is this a leftover test or an actual function?

File: eml/components/courses/CourseBody.jsx [hunk 15]
```diff
@@ -1,6 +1,6 @@
 import React from 'react'
 import { View } from 'react-native'
-import CourseSectionList from './courseBody/CourseSectionList'
+// TODO: Remove this - import CourseSectionList from './courseBody/CourseSectionList'
```
Comment (Jatewo) [inline eml/components/courses/CourseBody.jsx:3]: Is this supposed to be removed?

File: eml/components/courses/CourseHeader.jsx [hunk 16]
```diff
@@ -1,30 +1,34 @@
 import React from 'react'
-import { View, StyleSheet } from 'react-native'
+import { View, StyleSheet, Text } from 'react-native'
```
Comment (Jatewo) [inline eml/components/courses/CourseHeader.jsx:2]: Update to new Text component

File: eml/components/courses/CourseHeader.jsx [hunk 17]
```diff
@@ -1,30 +1,34 @@
...
+          progress={courseProgress}
+        >
+        </CourseTitleIcon>
+        {/*<DownloadCourseButton downloadStateSignal={downloadState} courseId={courseId} style={styles.downloadButton}></DownloadCourseButton>*/}
+      </View>
   )
 }
 
 
 const styles = StyleSheet.create({
```
Comment (Jatewo) [inline eml/components/courses/CourseHeader.jsx:30]: Please refactor to NativeWind as defined in our own guidelines

File: eml/components/courses/courseCard/CourseCard.jsx [hunk 18]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { View, Text, Image, Pressable, Dimensions } from 'react-native';
+import { useNavigation } from '@react-navigation/native';
+import { useFonts, VarelaRound_400Regular } from '@expo-google-fonts/dev';
```
Comment (Jatewo) [inline eml/components/courses/courseCard/CourseCard.jsx:4]: Not needed anymore

File: eml/components/courses/courseCard/CourseCard.jsx [hunk 19]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { View, Text, Image, Pressable, Dimensions } from 'react-native';
```
Comment (Jatewo) [inline eml/components/courses/courseCard/CourseCard.jsx:2]: Update to new Text component

File: eml/components/courses/courseCard/CourseCard.jsx [hunk 20]
```diff
@@ -0,0 +1,29 @@
...
+import { useNavigation } from '@react-navigation/native';
+import { useFonts, VarelaRound_400Regular } from '@expo-google-fonts/dev';
+import { AppLoading } from 'expo-app-loading';
+
+export default function CourseCard({ course, downloadState }) {
+    const navigation = useNavigation();
+    
+    let [fontsLoaded] = useFonts({
```
Comment (Jatewo) [inline eml/components/courses/courseCard/CourseCard.jsx:10]: Not needed anymore

File: eml/components/courses/courseCard/CourseCard.jsx (hunk 4 above)
Comment (Jatewo) [reply eml/components/courses/courseCard/CourseCard.jsx:15]: This is not needed, it is solved with the new Text component.

File: eml/components/courses/courseCard/CourseCard.jsx
Comment (Jatewo) [inline eml/components/courses/courseCard/CourseCard.jsx:1]: Use NativeWind

File: eml/components/courses/courseHeader/CourseProgress.jsx [hunk 21]
```diff
@@ -1,5 +1,5 @@
 import React from 'react'
-import { StyleSheet, View, Text } from 'react-native'
+import { StyleSheet, View, Text} from 'react-native'
```
Comment (Jatewo) [inline eml/components/courses/courseHeader/CourseProgress.jsx:2]: Update to new Text component

File: eml/components/courses/courseHeader/CourseProgress.jsx
Comment (Jatewo) [inline eml/components/courses/courseHeader/CourseProgress.jsx:1]: Use NativeWind

File: eml/components/courses/courseHeader/CourseTitleIcon.jsx (hunk 6 above)
Comment (Jatewo) [reply eml/components/courses/courseHeader/CourseTitleIcon.jsx:17]: Not needed (see earlier comments)

File: eml/components/courses/courseHeader/CourseTitleIcon.jsx [hunk 22]
```diff
@@ -3,49 +3,62 @@ import { StyleSheet, View, Image } from 'react-native'
 import Text from '../../general/Text';
 import PropTypes from 'prop-types'
 import { useFonts, VarelaRound_400Regular } from '@expo-google-fonts/dev'
```
Comment (Jatewo) [inline eml/components/courses/courseHeader/CourseTitleIcon.jsx:5]: Not needed

File: eml/components/courses/courseHeader/CourseTitleIcon.jsx
Comment (Jatewo) [inline eml/components/courses/courseHeader/CourseTitleIcon.jsx:1]: Use NativeWind

File: eml/components/courses/section/SectionCard.jsx
Comment (Jatewo) [inline eml/components/courses/section/SectionCard.jsx:1]: NativeWind

File: eml/components/courses/section/SectionCard.jsx [hunk 23]
```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react';
+import { View, Text, Pressable, Image, TouchableOpacity } from 'react-native';
```
Comment (Jatewo) [inline eml/components/courses/section/SectionCard.jsx:2]: Update to new Text component (won't comment on further files, but please change this for all 😄)

File: eml/components/easyDynComponents/easyDynCard.jsx
Comment (Jatewo) [inline eml/components/easyDynComponents/easyDynCard.jsx:1]: NativeWind (won't comment anymore about this but please fix for all files 😄)

File: eml/components/navBar/NavBar.jsx [hunk 24]
```diff
@@ -1,15 +1,163 @@
...
+
+const Tab = createBottomTabNavigator();
 
 export default function NavBar() {
+
+  let [fontsLoaded] = useFonts({
+    VarelaRound_400Regular
+  })
+
+  if (!fontsLoaded) {
```
Comment (Jatewo) [inline eml/components/navBar/NavBar.jsx:19]: Not needed

File: eml/components/navBar/NavBar.jsx [hunk 25]
```diff
@@ -1,15 +1,163 @@
...
+import { Icon } from '@rneui/themed';
+import { useFonts, VarelaRound_400Regular } from '@expo-google-fonts/dev'
+import { AppLoading } from 'expo-app-loading';
+
+const Tab = createBottomTabNavigator();
 
 export default function NavBar() {
+
+  let [fontsLoaded] = useFonts({
+    VarelaRound_400Regular
```
[the remark this answers is not in the document]
Comment (Jatewo) [reply eml/components/navBar/NavBar.jsx:16]: Not needed

File: eml/components/navBar/NavBar.jsx [hunk 26]
```diff
@@ -1,15 +1,163 @@
...
-import { NavigationContainer } from '@react-navigation/native'
-const Tab = createBottomTabNavigator()
+import React from 'react';
+import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
+import CourseScreen from '../../screens/courses/CourseScreen';
+import Explore from '../../screens/explore/Explore';
+import TestScreen from '../../screens/test/TestScreen';
+import ProfileComponent from '../../screens/profile/Profile';
+import { Icon } from '@rneui/themed';
+import { useFonts, VarelaRound_400Regular } from '@expo-google-fonts/dev'
```
Comment (Jatewo) [inline eml/components/navBar/NavBar.jsx:8]: Not needed

File: eml/components/navBar/NavBar.jsx
Comment (Jatewo) [inline eml/components/navBar/NavBar.jsx:1]: NativeWind

File: eml/package.json [hunk 27]
```diff
@@ -40,10 +40,11 @@
     "axios": "^0.27.2",
     "bootstrap": "^5.2.1",
     "expo": "^46.0.15",
-    "expo-app-loading": "^2.1.1",
+    "expo-app-loading": "~2.1.0",
```
Comment (Jatewo) [inline eml/package.json:43]: Are we supposed to use an older version?

**Review by Jatewo — CHANGES_REQUESTED** [review]
Overall good work. However, you are not using NativeWind in the files as described in our guidelines. Furthermore the new Text component should be used which handles the whole font import issue. Please respond to any comments if you are unsure about anything. Feel free to re-request my review when the requested changes has been addressed.

**Comment by JackSMunn** [comment]
Ping me when conflicts are resolved and checks pass, then I will review.

**Comment by JackSMunn** [comment]
Closing this pull request. As last commit was 10th october.

**Pull request closed without merging**